### PR TITLE
Update PostgresQL to version 13

### LINKED ISSUE #6050: postgres.dockerfile doesn't build
`toolset/databases/postgres/postgres.dockerfile` doesn't build anymore for me

Reporting before investigating

```
[12:48:56.471] Step 19/35 : RUN mv postgresql.conf /etc/postgresql/${PG_VERSION}/main/postgresql.conf
[12:48:56.943]  ---> Running in 0e1a5fb95a7f
[12:48:57.270] mv: cannot move 'postgresql.conf' to '/etc/postgresql/12/main/postgresql.conf': No such file or directory
[12:48:57.501] [STDERR] The command '/bin/sh -c mv postgresql.conf /etc/postgresql/${PG_VERSION}/main/postgresql.conf' returned a non-zero code: 1
[12:48:57.502]
```

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #6050 

Not sure this is the fix you want to take since it might impact benchmarks.

### CODE REVIEW

**Comment by volyrique** [comment]
Shouldn't we stop hardcoding version numbers? At the very least not inside the configuration files and not in the paths.

File: toolset/databases/postgres/postgres.dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-k
 RUN apt-get -yqq update > /dev/null
 RUN apt-get -yqq install locales
 
-ENV PG_VERSION 12
+ENV PG_VERSION 13
```
Comment (jcheron) [inline toolset/databases/postgres/postgres.dockerfile:17]: maybe add a `PG_VERSION` mask in `postgresql.conf` and replace it from here:
```bash
RUN sed -i "s|PG_VERSION|${PG_VERSION}|g" postgresql.conf
```

File: toolset/databases/postgres/postgresql.conf [hunk 2]
```diff
@@ -40,13 +40,13 @@
 
 data_directory = '/ssd/postgresql'		# use data in another directory
 					# (change requires restart)
-hba_file = '/etc/postgresql/12/main/pg_hba.conf'	# host-based authentication file
+hba_file = '/etc/postgresql/13/main/pg_hba.conf'	# host-based authentication file
```
Comment (jcheron) [inline toolset/databases/postgres/postgresql.conf:43]:
```
hba_file = '/etc/postgresql/PG_VERSION/main/pg_hba.conf'
```

**Comment by sebastienros** [comment]
@jcheron I think this PR solves it in a better way by pinning PG 12 at the source: https://github.com/TechEmpower/FrameworkBenchmarks/pull/6052

And in PG13 the default conf file is different, so I don't think making it a constant here makes sense, since bumping to13 should also change the conf file completely.

**Comment by jcheron** [comment]
PR https://github.com/TechEmpower/FrameworkBenchmarks/pull/6052 generates a 404 error on package downloads...
It is necessary to modify the list of packages without using the provided list file (which has the disadvantage of being also hard-coded).
```bash
RUN echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release  -sc || cat /etc/*-release|grep -oP  'CODENAME=\K\w+$'|head -1)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
```

**Pull request merged**